bodhi/server/services/updates.py

```diff
@@ -189,6 +194,36 @@ def set_request(request):
     return dict(update=update)
 
 
+@update_rerun_test.post(schema=bodhi.server.schemas.RerunTestSchema,
+                     validators=(
+                         validate_update_id,
+                         validate_acls,
+                     ),
+                     permission='edit', renderer='json',
+                     error_handler=bodhi.server.services.errors.json_handler)
+def rerun_test(request):
+    """
+    Send fedmsg request that a specific test for a specific update be re-run.
+
+    Args:
+        request (pyramid.request): The current request.
+    Returns:
+        dict: A dictionary with at least the following key mappings:
+            update: The update for which the rerun was requested.
+            test: The test for which the rerun was requested.
+            scenario: The scenario for which the rerun was requested.
+    """
+    data = request.validated
+    update = data['update']
+    test = data['test']
```

Comment:
Does the ```test``` here have any build names or update ids in it, or is it just the generic test name (like "dist.upgradepath")? If the latter, great! If the former, I'll point out that we may want to validate that the test being requested matches the update URL being posted to here.

Reply:
I *think* it's just the test name. That's what I'm expecting it to be, anyhow. We'll find out when I write some tests, I guess...